### PR TITLE
Closes GitHub issue #95: CG returns SUCCESS when not converged

### PR DESCRIPTION
**Pull request by anyzelman**
The CG when not converging previously returned SUCCESS, while FAILED was expected.

This MR fixes the detected issue, and also defines more clearly what a user may expect when different error codes are returned under different conditions.

No test was added due to algorithms at present only being smoke tested-- there are quite a few boundary conditions that proper unit tests should otherwise take into account.

### CODE REVIEW

**Comment by anyzelman** [comment]
Review OK. GitHub CI OK. Waiting internal CI OK and smoke tests + LPF OK.

File: include/graphblas/algorithms/conjugate_gradient.hpp [hunk 1]
```diff
@@ -85,7 +85,7 @@ namespace grb {
 		 * Additional outputs (besides \a x):
 		 *
 		 * @param[out]    iterations     The number of iterations the algorithm has
-		 *                               performed.
+		 *                               started.
```
Comment (djelovina) [inline include/graphblas/algorithms/conjugate_gradient.hpp:88]: Just a comment, if it is `out`, then is is number of iterations _performed_, is that right?

Comment (anyzelman) [reply include/graphblas/algorithms/conjugate_gradient.hpp:88]: If SUCCESS were returned yes -- otherwise, it is possible that an iteration is stopped mid-way, hence the change

**Comment by anyzelman** [comment]
Internal CI OK, smoke tests + LPF OK, will merge
